dirty_cat/gap_encoder.py

```diff
@@ -540,6 +433,190 @@ def transform(self, X):
+                 analyzer='char', add_words=False, random_state=None,
+                 rescale_W=True, max_iter_e_step=20):
+
+        self.ngram_range = ngram_range
+        self.n_components = n_components
+        self.gamma_shape_prior = gamma_shape_prior  # 'a' parameter
+        self.gamma_scale_prior = gamma_scale_prior  # 'b' parameter
+        self.rho = rho
+        self.rho_ = self.rho
+        self.rescale_rho = rescale_rho
+        self.batch_size = batch_size
+        self.tol = tol
+        self.hashing = hashing
+        self.hashing_n_features = hashing_n_features
+        self.max_iter = max_iter
+        self.min_iter = min_iter
+        self.init = init
+        self.analyzer = analyzer
+        self.add_words = add_words
+        self.random_state = random_state
+        self.rescale_W = rescale_W
+        self.max_iter_e_step = max_iter_e_step
+
+    # TODO: score method
+
+    def _create_column_gap_encoder(self) -> GapEncoderColumn:
+        return GapEncoderColumn(
+            ngram_range=self.ngram_range,
+            n_components=self.n_components,
+            gamma_shape_prior=self.gamma_shape_prior,
+            gamma_scale_prior=self.gamma_scale_prior,
+            rho=self.rho,
+            rescale_rho=self.rescale_rho,
+            batch_size=self.batch_size,
+            tol=self.tol,
+            hashing=self.hashing,
+            hashing_n_features=self.hashing_n_features,
+            max_iter=self.max_iter,
+            init=self.init,
+            add_words=self.add_words,
+            random_state=self.random_state,
+            rescale_W=self.rescale_W,
+            max_iter_e_step=self.max_iter_e_step,
+        )
+
+    def fit(self, X, y=None):
+        X = check_x(X)
+        self.fitted_models_ = []
+        for k in range(X.shape[1]):
+            ge_column = self._create_column_gap_encoder()
+            self.fitted_models_.append(ge_column.fit(X[:, k]))
+        return self
+
+    def transform(self, X):
+        # Check input data shape
+        X = check_x(X)
+        X_enc = []
+        for k in range(X.shape[1]):
+            X_enc.append(self.fitted_models_[k].transform(X[:, k]))
+        X_enc = np.hstack(X_enc)
+        return X_enc
+
+    def partial_fit(self, X, y=None):
+        X = check_x(X)
+        # Init the `GapEncoderColumn` instances if the model was
+        # not fitted already.
+        if not hasattr(self, 'fitted_models_'):
+            self.fitted_models_ = [
+                self._create_column_gap_encoder()
+                for _ in range(X.shape[1])
+            ]
+
+        for k in range(X.shape[1]):
+            self.fitted_models_[k].partial_fit(X[:, k])
+
+        return self
+
+    def get_feature_names(self, n_labels=3):
+        assert hasattr(self, 'fitted_models_'), (
+            'ERROR: GapEncoder must be fitted first.')
+        labels = [
+            enc.get_feature_names(n_labels)
+            for enc in self.fitted_models_]
```

Comment:
Good idea, now the labels have a prefix which is either the actual column name (if the input is a dataframe), or `'col{k}'`, with k the column index otherwise.